orttraining/orttraining/test/graph/gradient_graph_builder_test.cc

```diff
@@ -1115,7 +1115,107 @@ void RetrieveSendRecvOperators(
   }
 }
 
-TEST(GradientGraphBuilderTest, PipelineOnlinePartition) {
+TEST(GradientGraphBuilderTest, PipelineOnlinePartition_bert_tiny) {
+  const auto model_path = ORT_TSTR("testdata/bert-tiny.onnx");
+
+  const size_t total_partition_count = 3;
+  TrainingSession::TrainingConfiguration::PipelineConfiguration pipe{};
+  pipe.do_partition = true;
+
+  // evenly model in 3 partitions
+  TrainingSession::TrainingConfiguration::CutInfo cut0 = {
+      onnxruntime::training::TrainingSession::TrainingConfiguration::CutEdge("186"),
```

Comment:
Do we have test cases for invalid cuts?  (IDs not present at all, or invalid in other ways).  Does the ordering of the IDs matter in the list?  If so then we should test that also if it is not already here.

Reply:
We do now :-)

---
In reply to: [452243541](https://github.com/microsoft/onnxruntime/pull/4321#discussion_r452243541) [](ancestors = 452243541)